core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeDDLBuilder.scala

```diff
@@ -219,4 +218,38 @@ class SnowflakeDDLBuilder
       affectedColumns.map(col => ir.DropConstraint(Some(col), constraint))
     }
   }
+
+  private def buildVariables(ctx: DeclareContext): Seq[ir.SetVariable] = {
+    val ids = ctx.id().asScala.map(_.getText)
+    val dataTypes = ctx.dataType().asScala.map(dt => Some(DataTypeBuilder.buildDataType(dt)))
+    val expressions = ctx.expr().asScala.map(_.accept(expressionBuilder)).map(Some(_))
+
+    // Padding the lists to ensure they all have the same size
+    // [TODO]: This is a temporary fix. We need to handle this in a better way
+    val paddedDataTypes = dataTypes ++ Seq.fill(ids.size - dataTypes.size)(None)
+    val paddedExpressions = (expressions ++ Seq.fill(ids.size - expressions.size)(None))
+
+    ids.zip(paddedDataTypes).zip(paddedExpressions).map { case ((name, dataType), expr) =>
```

Comment:
Whenever you find yourself needing to write this kind of complicated logic it is a sign that the corresponding grammar could be improved to make your life easier (see the comments above in `SnowflakeParser.g4`.